### PR TITLE
add checkpoint for cluster capabilities

### PR DESCRIPTION
**Pull request by jinyunma**
Add checkpoint for cluster capabilties

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei @yunjiang29 to review.
I add step in test part to check capabilities on different configurations.
If that is okay, plan to update relevant job config(all versions, 4.11+) to insert this step.

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  if [ ${release_image} != "" ]; then
+    oc registry login
+    version=$(oc adm release info ${release_image} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+  fi
+  echo "${version}"
+}
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
```
Comment (yunjiang29) [inline ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:30]: As following steps require to access cluster, I would suggest to exit directly if `${SHARED_DIR}/kubeconfig` not found.

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+check_result=0
+# Enabled capabilities check
+echo "------check enabled capabilities-----"
+echo "enabled_operators: ${enabled_operators}"
+for op in $enabled_operators; do
+    stdout=$(mktemp)
+    stderr=$(mktemp)
+    oc get co ${caps_operator[$op]} 1>${stdout} 2>${stderr} || true
+    if [[ -s "${stderr}" ]]; then
+        echo -e "ERROR: some operators are not deployed:\n"
```
[the remark this answers is not in the document]
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:120]: not only one operator for each capability.
For example, capability `MachineAPI` contains three operators.
```
caps_operator[MachineAPI]="machine-api control-plane-machine-set cluster-autoscaler"
```

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+oc get co
+
+check_result=0
+# Enabled capabilities check
+echo "------check enabled capabilities-----"
+echo "enabled_operators: ${enabled_operators}"
+for op in $enabled_operators; do
+    stdout=$(mktemp)
+    stderr=$(mktemp)
+    oc get co ${caps_operator[$op]} 1>${stdout} 2>${stderr} || true
```
[the remark this answers is not in the document]
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:118]: good suggestion, let me update.

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (yunjiang29) [reply ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:120]: ahh, get it

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        fi
+    done
+done
+
+# Disabled capabilities check
+echo "------check disabled capabilities-----"
+echo "disabled capability set: ${disabled_capability_set}"
+for cap in $disabled_capability_set; do
+    for op in ${caps_operator[$cap]}; do
+        if [[ `grep "${op}" ${co_content}` ]]; then
```
Comment (yunjiang29) [inline ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:165]:
```suggestion
        if [[ `grep -e "^${op} " ${co_content}` ]]; then
```

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+co_content=$(mktemp)
+oc get co | tee ${co_content}
+
+check_result=0
+# Enabled capabilities check
+echo "------check enabled capabilities-----"
+echo "enabled capability set: ${enabled_capability_set}"
+for cap in $enabled_capability_set; do
+    for op in ${caps_operator[$cap]}; do
+        if [[ ! `grep "${op}" ${co_content}` ]]; then
```
Comment (yunjiang29) [inline ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:153]: @jinyunma It's better to match the operator's name more accurately:
e.g. we don't to match `operator-lifecycle-manager-catalog` if given `operator-lifecycle-manager`
```
operator-lifecycle-manager
operator-lifecycle-manager-catalog
operator-lifecycle-manager-packageserver
```


Note: there is a space(` `) after `${op}`
```suggestion
        if [[ ! `grep -e "^${op} " ${co_content}` ]]; then
```

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh (hunk 4 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh]: Fixed

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh (hunk 5 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh]: fixed

**Pull request merged**